capstone/src/test/java/com/google/sps/servlets/AnnouncementTest.java

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** */
+@RunWith(JUnit4.class)
+public final class AnnouncementTest {
+
+  private static final String ANNOUNCEMENT_PROP = "Announcement";
+  private static final String AUTHOR_PROP = "author";
+  private static final String TIME_PROP = "time";
+  private static final String CONTENT_PROP = "content";
+  private static final String CLUB_PROP = "club";
+
+  private static final String AUTHOR_1 = "kshao@google.com";
+  private static final String AUTHOR_2 = "kakm@google.com";
+  private static final String AUTHOR_3 = "meganshi@google.com";
+
+  private static final String CLUB_1 = "Club 1";
+  private static final String CLUB_2 = "Club 2";
+  private static final String CLUB_3 = "Club 3";
+  private static final String CONTENT_1 = "Cool test comment.";
+
+  private static final long TIME_10 = 10L;
```

Comment:
A couple of notes of testing constants.

1. Are these the same as the ones in the Constants file? If so, you can call those directly instead of defining them here. If not, ignore this.
2. If these constants are specific to the individual test use cases, it's actually better to define them or even hard-code them in the specific test. Test constants are generally not used except for commonly shared cases (ex: the email of the user is always going to return the same thing). If it's for set-up though, it's ok to leave it as-is.

Reply:
I'm a bit confused about this comment - currently, I'm using these constants in setUp() to add them to Datastore, and then using them again to verify that they're what the servlet return from Datastore. Does that mean they're specific to individual test use cases, or best left like this?

Reply:
Sorry, this is fine because these are needed for the setup. I just mean in general, if you're writing that creates a new object (using constants as fields in the constructor for example), and that object is only used inside a specific test, then we don't need those constants to be defined outside the test as constants.

Btw, doesn't seem like all of these constants are used.